src/lib/files/ProjectFilesManager.ts

```diff
@@ -0,0 +1,75 @@
+import { createReadStream, lstat, readFileSync, statSync } from 'fs-extra'
+import * as glob from 'globby'
+import { join } from 'path'
+import { reject } from 'ramda'
+
+export const pathToFileObject = (root: string, prefix: string = '') => {
+  return (path: string): BatchStream => {
+    const realAbsolutePath = join(root, path)
+    const stats = statSync(realAbsolutePath)
+    return {
+      path: join(prefix, path),
+      content: createReadStream(realAbsolutePath),
+      byteSize: stats.size,
+    }
+  }
+}
+
+export class ProjectFilesManager {
+  public static DEFAULT_IGNORED_FILES = [
+    '.DS_Store',
+    'README.md',
+    '.gitignore',
+    'package.json',
+    'node_modules/**',
+    '**/node_modules/**',
+    '.git/**',
+  ]
+
+  private static isTestOrMockPath = (path: string) => {
+    return /.*(test|mock|snapshot).*/.test(path.toLowerCase())
+  }
+
+  public root: string
+  constructor(projectRoot: string) {
+    this.root = projectRoot
+  }
+
+  private getIgnoredPaths(test: boolean = false): string[] {
+    try {
+      const filesToIgnore = readFileSync(join(this.root, '.vtexignore'))
+        .toString()
+        .split('\n')
+        .map(path => path.trim())
+        .filter(path => path !== '')
+        .map(path => path.replace(/\/$/, '/**'))
+        .concat(ProjectFilesManager.DEFAULT_IGNORED_FILES)
+      return test ? reject(ProjectFilesManager.isTestOrMockPath, filesToIgnore) : filesToIgnore
+    } catch (e) {
+      return ProjectFilesManager.DEFAULT_IGNORED_FILES
+    }
+  }
+
+  public async getLocalFiles(test: boolean = false): Promise<string[]> {
```

Comment:
Can you clarify what is it for? Where is it used?

Reply:
`ProjectFilesManager` isn't used yet :(
The `vtex link` code is somewhat difficult to refactor, so I'm doing it in parts - The `ProjectFilesManager` will replace functions on `file.ts` regarding getting the files to send to builder-hub